### ISSUE TITLE
Apply decimals to invocation parameters

### ISSUE THREAD

**Issue by piyalbasu** [stellar/freighter#1511]
Sometimes we get back invocation parameters that require decimal formatting. How can we accurately figure out if the param needs to be formatted when we render?

![Image](https://github.com/user-attachments/assets/b28bbcd3-99b6-4666-8a76-4571e5807d2a)

**Comment by aristidesstaffieri** [comment]
I think that we cannot format all int/uint values that we encounter from an invoke host function because in cases like the screenshot we don't have an easy way to know what they represent generically.

How do we know what `stake_amount` represents in this case?
I think we would have to understand the protocol architecture to then be able to correlate that uint to a balance on a token which then can be formatted by its decimals.

We can add special cases for some known and frequently used protocols. It would be nice to see a formatted token balance value in the screenshot above. 

Am I missing anything about this?

@stellar/wallet-eng

**Comment by JakeUrban** [comment]
I agree with that approach, and it sounds like a common theme across a variety of UX-related items.

For example, right now we're displaying contract calls as "invoke host function" in the history tab. I don't know how much we'll be able to improve this for all contract calls generically, but we can provide a better UX for a set of known contracts that we add support for explicitly.

Maybe we should create some kind of checklist for all the ways in which we'd want to improve UX when adding a new protocol to the set.

**Comment by aristidesstaffieri** [comment]
> I agree with that approach, and it sounds like a common theme across a variety of UX-related items.
> 
> For example, right now we're displaying contract calls as "invoke host function" in the history tab. I don't know how much we'll be able to improve this for all contract calls generically, but we can provide a better UX for a set of known contracts that we add support for explicitly.
> 
> Maybe we should create some kind of checklist for all the ways in which we'd want to improve UX when adding a new protocol to the set.

I can put together some notes on where we can take advantage of special cases for well known contracts and then a list of initial contracts to consider for special cases.

I think also, more long term, maybe this can also be solved with a SEP that defined how an issuer of a protocol or set of contracts can host metadata on how to display or format certain values in wallets or other UIs.

**Comment by aristidesstaffieri** [comment]
Ok so the places where we can take advantage of a special-case list of contract parameter formats are in -
/account-history
/sign-transaction
/sign-auth-entry

In account history, we already do a special case for any SAC or Sep41 token operations.
We identify SAC instances by making sure the `name` field on the contract is derived from the corresponding asset issuer.
We identify Sep41 tokens by fetching the contract's spec and checking that the interface aligns with the interface in Sep41.

On both of the signing workflows, we do not attempt to format any parameters at this point.

Some initial improvements could include - 
1. Treating parameters on signing screens in a similar way to the account history items for SAC or Sep41 token operations(format using decimals).
2. Expand operation labeling in account history. We special case token transfers but we could more generically fetch the contract spec and improve the labels on any operation on any contract in a similar way to the signing flows.
3. Create special cases for common contracts. For example - label a Phoenix stake amount according to underlying token decimals.

**Comment by JakeUrban** [comment]
Agreed on all points, though this one warrant more product input:

> Expand operation labeling in account history. We special case token transfers but we could more generically fetch the contract spec and improve the labels on any operation on any contract in a similar way to the signing flows.

Maybe we should be focusing more on the effects of calling a transaction, rather than the specific function called. For example, Blend's contract interface doesn't include many human-friendly function names, but the balances changes make it clear what happened.

**Comment by JakeUrban** [comment]
@brunomuler @sdfcharles curious on your takes here

**Comment by sdfcharles** [comment]
@JakeUrban agreed with the above comments, that we need to standardize labels and apply decimals where we can. ideally, we also can showcase the dollar (or set local currency in freighter. this is not built yet but would be great to have as an option) amount equivalent next to it as well to reinforce that it's a balance.

**Comment by JakeUrban** [comment]
Should this issue still be in QA? What are the next steps?

**Comment by aristidesstaffieri** [comment]
Yes I will move this to done, and I think these are the next steps:
1. Align special case parameter handling across screens.
2. Decide which contracts we will special case for labeling, and come up with a per contract strategy to render details about their operations.

USD values are tracked as a separate spike so will leave that off of here.

Let me know if I've missed anything.
